src/algorithms/optimization/fixed_point_differentiation.jl

```diff
@@ -111,6 +111,7 @@ function _rrule(
     alg::CTMRGAlgorithm,
 )
     env, info = leading_boundary(envinit, state, alg)
+    @reset alg.projector_alg.trscheme = FixedSpaceTruncation() # fix spaces during differentiation
```

Comment:
I'm not too familiar with `Accessors`, but can you make this create a new variable (with a new variable name) instead? This usually ruins the type stability of the closure below